test/lib/validation/ValidationError.js

```diff
@@ -0,0 +1,938 @@
+		"analyzeYamlError should return expected results");
+});
+
+test.serial("ValidationError.analyzeYamlError: Invalid documentIndex=1 (With leading separator)", (t) => {
+	const error = {dataPath: "/property3"};
+	const yaml = {
+		path: "/my-project/ui5.yaml",
+		source:
+`---
+property1: value1document1
+property2: value2document1
+property3: value3document1
+property4: value4document1
+property5: value5document1
+`,
+		documentIndex: 1
+	};
+
+	const info = ValidationError.analyzeYamlError({error, yaml});
+
+	t.deepEqual(info, {line: -1, column: -1},
+		"analyzeYamlError should return expected results");
+});
+
+test.serial("ValidationError.analyzeYamlError: Invalid documentIndex=1 (Without leading separator)", (t) => {
+	const error = {dataPath: "/property3"};
+	const yaml = {
+		path: "/my-project/ui5.yaml",
+		source:
+`property1: value1document1
+property2: value2document1
+property3: value3document1
+property4: value4document1
+property5: value5document1
+`,
+		documentIndex: 1
+	};
+
+	const info = ValidationError.analyzeYamlError({error, yaml});
+
+	t.deepEqual(info, {line: -1, column: -1},
+		"analyzeYamlError should return expected results");
+});
+
+test.serial("ValidationError.formatMessage: keyword=type dataPath=", (t) => {
+	const error = {
+		dataPath: "",
+		keyword: "type",
+		message: "should be object",
+		params: {
+			type: "object",
+		},
+		schemaPath: "#/type",
+	};
+
+	const expectedErrorMessage = "Configuration should be of type 'object'";
+
+	const errorMessage = ValidationError.formatMessage(error);
+	t.is(errorMessage, expectedErrorMessage);
+});
+
+test.serial("ValidationError.formatMessage: keyword=type", (t) => {
+	const error = {
+		dataPath: "/foo",
+		keyword: "type",
+		message: "should be object",
+		params: {
+			type: "object",
+		},
+		schemaPath: "#/type",
+	};
+
+	const expectedErrorMessage = `Configuration ${chalk.underline(chalk.red("foo"))} should be of type 'object'`;
+
+	const errorMessage = ValidationError.formatMessage(error);
+	t.is(errorMessage, expectedErrorMessage);
+});
+
+test.serial("ValidationError.formatMessage: keyword=required w/o dataPath", (t) => {
+	const error = {
+		dataPath: "",
+		keyword: "required",
+		message: "should have required property 'specVersion'",
+		params: {
+			missingProperty: "specVersion",
+		},
+		schemaPath: "#/required",
+	};
+
+	const expectedErrorMessage = "Configuration should have required property 'specVersion'";
+
+	const errorMessage = ValidationError.formatMessage(error);
+	t.is(errorMessage, expectedErrorMessage);
+});
+
+test.serial("ValidationError.formatMessage: keyword=required", (t) => {
+	const error = {
+		keyword: "required",
+		dataPath: "/metadata",
+		schemaPath: "../ui5.json#/definitions/metadata/required",
+		params: {missingProperty: "name"},
+		message: "should have required property 'name'"
+	};
+
+	const expectedErrorMessage = `Configuration ${chalk.underline(chalk.red("metadata"))} should have required property 'name'`;
```

Comment:
"Configuration ${chalk.underline(chalk.red("metadata"))} must have the required property 'name'"